pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -122,8 +122,14 @@ func (c *Controller) Finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 }
 
 func (c *Controller) ensureTerminationGracePeriodExpirationAnnotation(ctx context.Context, node *v1.Node, nodeClaim *v1beta1.NodeClaim) error {
-	if nodeClaim.Spec.TerminationGracePeriod != nil && nodeClaim.ObjectMeta.DeletionTimestamp != nil {
-		expirationTimeString := nodeClaim.DeletionTimestamp.Time.Add(nodeClaim.Spec.TerminationGracePeriod.Duration).Format(time.RFC3339)
+
+	nodePool, err := nodeclaimutil.NodePoolForNodeClaim(ctx, c.kubeClient, nodeClaim)
```

Comment:
Should we short circuit here if the annotation is already set? Realistically we list NodePools frequently enough that I would imagine they are cached 90% of the time but it seems reasonable to short circuit anyway.